PP_APP/server/frontend/static/js/script.js

```diff
@@ -3,126 +3,87 @@ let user_email = document.getElementById('user_email');
 let top_input = document.getElementById('top_img');
 let side_input = document.getElementById('side_img');
 
-function isEmail(email) {
-    var regex = /^([a-zA-Z0-9_.+-])+\@(([a-zA-Z0-9-])+\.)+([a-zA-Z0-9]{2,4})+$/;
-    return regex.test(email);
-}
-function checkIfImageIsInvalid(img){
-  if (img === undefined || !validImageTypes.includes(img['type'])) {
-    return true;
-  }
-  return false;
-}
-// Display image before uploading
-// https://stackoverflow.com/questions/4459379/preview-an-image-before-it-is-uploaded
-function readURL(input, elem_id) {
-  if (input.files && input.files[0]) {
-    var reader = new FileReader();
-    reader.onload = function (e) {
-      $('#' + elem_id).attr('style', 'background-image: url(' + e.target.result + ');background-repeat:no-repeat;background-size:cover;color:#fff;background-attachment:scroll;background-position:center;');
-    }
+let base_url = window.location.origin;
+let host = window.location.host;
+let pathArray = window.location.pathname.split('/');
+
+let GV_HOST = "http://127.0.0.1:8080";
```

Comment:
This is a local GV host; there is a dedicated server for this, the host is 34.71.6.144. We agreed that Flask and Django should not run on the same server due to possible traffic issues, so I spent a decent amount of time configuring a VM to serve the Flask app. I think we should stick to the architecture we outlined because the infrastructure for the Flask app running on a separate VM has already been built.

All documentation is in the flask_app directory, and I have already verified that it works when writing my own ajax calls.

Reply:
The Javascript is something that is downloaded to the client-side by the browser before it executes. Therefore, It has nothing to do with Django or Flask. I installed Flask locally and it runs on a different port than Django (which runs on 8000) for testing purposes. And also because it is better for cost reasons as each server call accumulates cost. This is perfectly in accordance with our architecture.

Reply:
We can host flask on on the same VM as Django, but for the record I have been hosting the flask app on Google Cloud constantly for the past month to make it available for everyone because that is what our architecture outlined. If this wasn't the plan or a part of our architecture, I should have been told when I submitted my PR's that emphasized that it's hosted on it's own VM to save money and time.